ocs_ci/ocs/resources/machineconfig.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf8 -*-
+
+
+import logging
+
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.objectconfigfile import ObjectConfFile
+
+
+logger = logging.getLogger(__name__)
+
+
+def deploy_machineconfig(tmp_path, mc_name, mc_dict, mcp_num=2):
+    """
+    Deploy given MachineConfig dict and wait for the configuration to be
+    deployed on all MachineConfigPools.
+    """
+    # deploy the machine config within openshift-config namespace
+    mc_file = ObjectConfFile(mc_name, mc_dict, None, tmp_path)
+    mc_file.create(namespace="openshift-config")
+    # now let's make sure the MCO (machine config operator) noticed just
+    # deployed givne machine config and started to process it
+    logger.info(
+        "waiting for both machineconfigpools to be updating "
+        "as a result of deployment of given machineconfig"
+    )
+    mcp_h = OCP(kind="MachineConfigPool", namespace="openshift-config")
+    mcp_h.wait_for_resource(
+        resource_count=mcp_num,
+        condition="True",
+        column="UPDATING",
+        sleep=5,
+        timeout=120,
+    )
+    # and now wait for MachineConfigPools to be updated and ready
+    logger.info("waiting for %d machineconfigpools to be updated and ready", mcp_num)
+    mcp_h.wait_for_resource(
+        resource_count=mcp_num,
+        condition="True",
+        column="UPDATED",
+        sleep=60,
+        timeout=1800,
+    )
+    # also check that no pools are degraded
+    mcp_h.wait_for_resource(
+        resource_count=mcp_num,
+        condition="False",
+        column="DEGRADED",
+        sleep=10,
+        timeout=120,
+    )
+    logger.info("MachineConfig %s has been deployed", mc_name)
```

Comment:
Just FYI, there already is [`wait_for_machineconfigpool_status(...)`](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/utility/utils.py#L3540) function, but your approach is more complex, so I think it is ok to not replace it with that function (and we might think about changing the original function to do it this way).